components/shared_code/src/shared_data_model/meta/entity.py

```diff
@@ -58,7 +58,7 @@ class EntityAttribute(NamedModel):
     def set_key(self) -> Self:
         """Set the key to the lower case version of the name if there's no key."""
         if self.key is None:
-            self.key = self.name.lower().replace(" ", "_")
+            self.key = self.name.lower().replace(" ", "_").replace("-", "_")
```

Comment:
Are there no attributes that need to be migrated for this, or are they mapped automatically between versions (i.e. when time travelling or comparing with older dates which are on older datamodel format)?

Reply:
No, I checked. This is the first time we use hyphens in attribute names.